handler/src/main/java/io/netty/handler/stream/ChunkedWriteHandler.java

```diff
@@ -167,21 +167,24 @@ private void discard(Throwable cause) {
             if (message instanceof ChunkedInput) {
                 ChunkedInput<?> in = (ChunkedInput<?>) message;
                 try {
-                    if (!in.isEndOfInput()) {
+                    // read state of the input in local variables before closing it
+                    boolean endOfInput = in.isEndOfInput();
+                    long inputLength = in.length();
+                    closeInput(in);
+                    if (!endOfInput) {
                         if (cause == null) {
                             cause = new ClosedChannelException();
                         }
                         currentWrite.fail(cause);
                     } else {
-                        currentWrite.success(in.length());
+                        currentWrite.success(inputLength);
```

Comment:
@lutovich Not related to this specific change, but... I wonder if there's a chance that the write was already set to success/failure 🤔 Here's [a comment](https://github.com/netty/netty/blob/33c18469349d45b139506446f356e2b4f3af7086/handler/src/main/java/io/netty/handler/stream/ChunkedWriteHandler.java#L217-L225) from `doFlush`, not sure if the same is applicable here.

Reply:
@kachayev I'm not sure, to be honest. `currentWrite` is a `PendingWrite` and uses `trySuccess/tryFailure` to complete the underlying promise. It shouldn't be a problem to invoke `success/failure` more than one, right?

Reply:
Right, should not be a problem. I'm just trying to understand if closing the input (potentially?) twice will cause any troubles.

Reply:
I think it should be ok... if not we can do another PR.